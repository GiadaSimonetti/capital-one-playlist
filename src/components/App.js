import React, { Component } from "react";

class App extends Component {
  render(){
    return (
      <div className="playlist">
        <h1>The Playlist</h1>
        <div className="playlist-table-div">

        <table id="playlist-table">
          <thead>
            <tr>
              <th>Title</th>
              <th>Artist</th>
              <th>Album</th>
              <th>Year</th>
            </tr>
          </thead>
          <tbody>
            
          </tbody>
        </table>

        </div>
      </div>
    );
  }
}

export default App;
