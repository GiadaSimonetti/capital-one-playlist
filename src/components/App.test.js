import React from "react";
import { shallow } from "enzyme";
import setupTests from './../setupTests.js';
import tempPolyfills from './../tempPolyfills';
import App from "./App";

const app = shallow(<App />);

it("renders correctly", () => {
  expect(app).toMatchSnapshot();
});

it("initializes the size in `state`", () => {
    expect(app.state()).toEqual({ size: 6});
  });

describe("table", () =>{
  it("returns the number of `TableHeaderColumn`", () => {
    expect(app.find(".playlist-table").children().length).toEqual(4);
  });
  // 
  // it("checks the existence of `Title` column", () => {
  //   expect(app.find(".title-field").exists()).toBe(true)
  // });

  it("checks the existence of `Title` column", () => {
    expect(app.find(".title-field").props().children).toEqual("Title");
  });

});
