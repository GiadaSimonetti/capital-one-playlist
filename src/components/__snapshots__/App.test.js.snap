// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          The Playlist
</h1>,
        <div
          className="playlist-table-div"
>
          <table
                    id="playlist-table"
          >
                    <thead>
                              <tr>
                                        <th>
                                                  Title
                                        </th>
                                        <th>
                                                  Artist
                                        </th>
                                        <th>
                                                  Album
                                        </th>
                                        <th>
                                                  Year
                                        </th>
                              </tr>
                    </thead>
                    <tbody />
          </table>
</div>,
      ],
      "className": "playlist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Playlist",
        },
        "ref": null,
        "rendered": "The Playlist",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <table
            id="playlist-table"
>
            <thead>
                        <tr>
                                    <th>
                                                Title
                                    </th>
                                    <th>
                                                Artist
                                    </th>
                                    <th>
                                                Album
                                    </th>
                                    <th>
                                                Year
                                    </th>
                        </tr>
            </thead>
            <tbody />
</table>,
          "className": "playlist-table-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                Title
                                </th>
                                <th>
                                                Artist
                                </th>
                                <th>
                                                Album
                                </th>
                                <th>
                                                Year
                                </th>
                </tr>
</thead>,
              <tbody />,
            ],
            "id": "playlist-table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                    Title
                  </th>
                  <th>
                                    Artist
                  </th>
                  <th>
                                    Album
                  </th>
                  <th>
                                    Year
                  </th>
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Title
</th>,
                    <th>
                      Artist
</th>,
                    <th>
                      Album
</th>,
                    <th>
                      Year
</th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title",
                    },
                    "ref": null,
                    "rendered": "Title",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Artist",
                    },
                    "ref": null,
                    "rendered": "Artist",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Album",
                    },
                    "ref": null,
                    "rendered": "Album",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Year",
                    },
                    "ref": null,
                    "rendered": "Year",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The Playlist
</h1>,
          <div
            className="playlist-table-div"
>
            <table
                        id="playlist-table"
            >
                        <thead>
                                    <tr>
                                                <th>
                                                            Title
                                                </th>
                                                <th>
                                                            Artist
                                                </th>
                                                <th>
                                                            Album
                                                </th>
                                                <th>
                                                            Year
                                                </th>
                                    </tr>
                        </thead>
                        <tbody />
            </table>
</div>,
        ],
        "className": "playlist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Playlist",
          },
          "ref": null,
          "rendered": "The Playlist",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <table
              id="playlist-table"
>
              <thead>
                            <tr>
                                          <th>
                                                        Title
                                          </th>
                                          <th>
                                                        Artist
                                          </th>
                                          <th>
                                                        Album
                                          </th>
                                          <th>
                                                        Year
                                          </th>
                            </tr>
              </thead>
              <tbody />
</table>,
            "className": "playlist-table-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                                    <th>
                                                      Title
                                    </th>
                                    <th>
                                                      Artist
                                    </th>
                                    <th>
                                                      Album
                                    </th>
                                    <th>
                                                      Year
                                    </th>
                  </tr>
</thead>,
                <tbody />,
              ],
              "id": "playlist-table",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th>
                                        Title
                    </th>
                    <th>
                                        Artist
                    </th>
                    <th>
                                        Album
                    </th>
                    <th>
                                        Year
                    </th>
</tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                        Title
</th>,
                      <th>
                        Artist
</th>,
                      <th>
                        Album
</th>,
                      <th>
                        Year
</th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Title",
                      },
                      "ref": null,
                      "rendered": "Title",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Artist",
                      },
                      "ref": null,
                      "rendered": "Artist",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Album",
                      },
                      "ref": null,
                      "rendered": "Album",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Year",
                      },
                      "ref": null,
                      "rendered": "Year",
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
